google-cloud-pubsub/test/google/cloud/pubsub/async_publisher_test.rb

```diff
@@ -344,6 +344,8 @@
 
     callbacks.must_equal 2
 
+    sleep 0.1
```

Comment:
`sleep` statement in async tests is still flaky, because it doesn't guarantee the child thread runs during the sleep. It's better to use a blocking check to verify some sort of state, such as [this](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/ca518c824e0db4070148604fe1f66a35bd204a51/stackdriver-core/test/stackdriver/core/async_actor_test.rb#L241-L243).